### DISCUSSION TITLE
Preload of large number of audio files fails

### DISCUSSION THREAD

**Discussion by christinalubinus** [jspsych/jsPsych#1286] [Q&A] [answered]
Dear all, 

in my experiment, I plan to present a large number of audio files (~300). While creating the experiment, I was working with just 4 example files and the code was running well. Now, when testing with all files, the experiment fails to load, even when the `max_load_time` is increased. I saw that there are some threads about this topic already but non could quite answer my question, thus I decided to open a new one. 

Sorry for the lengthy post, but here is some more detail: I load a js file with all stimulus information (see below). Then I prepare and load all audios following the code from the semantic relatedness example. 

```javascript
// snippet from the .js stimulus file 
var comprehen_trial_info_v1 = [
  {
    stimnum: 1,
    file: "sound/comprehension/stimuli/s2_29_fast.wav",
    compr_factor: "1,942973793",
    freq: "3,5",
    stim_ID: 29,
    speaker: 2
  },
// many more trials here... 
]

// the following is taken from the main file
// preparing the preload of audio
var main_task_audio = [];
  for (var i=0; i<comprehen_trial_info_v1.length; i++) {
      main_task_audio.push(comprehen_trial_info_v1[i].file);
  }


// ...some trials defined here...

// critical trial for presenting audio
var audio = {
    type: 'audio-keyboard-response',
    stimulus: jsPsych.timelineVariable('file'),
    prompt: '<span style="font-size:40px;">+</span>',
    choices: jsPsych.NO_KEYS,
    trial_ends_after_audio: true,
    data: {
        freq: jsPsych.timelineVariable('freq'),
      //  stimulus: jsPsych.timelineVariable('stimulus_ID')

    },
    on_finish: function() {
      trial_count++;
    }
}

// procedure 
var main_procedure = {
    timeline: [fixation, audio, question1,question2, pause_conditional],
    timeline_variables: comprehen_trial_info_v1,
    randomize_order: true
 };
 timeline1.push(main_procedure);

// preload as part of jspsych.init()
jsPsych.init({
        timeline: timeline1,
        preload_audio: [main_task_audio],
        show_preload_progress_bar: true,
        use_webaudio: false,
        max_load_time: 12000000,
       
      })
```

The error I get is the following:  
<img width="1304" alt="error1" src="https://user-images.githubusercontent.com/45943148/101280126-87206980-37c7-11eb-95a0-8814dcdcf1ee.png">

By following a former thread (https://github.com/jspsych/jsPsych/discussions/739), I added a separate preload trial. This did change the behavior, as the experiment does not crash right upon opening the link. But as soon as it gets to the first audio trial, it freezes. 
```javascript
  var preload_audio = {
      type: "call-function",
      async: true,
      func: function (done) {
          jsPsych.getDisplayElement().innerHTML = '<p> Loading... </p>'
          jsPsych.pluginAPI.preloadAudioFiles([jsPsych.timelineVariable('file',true)], function () {
              done({ preload: "success" });
          })
      }
  };
```

<img width="1164" alt="error2" src="https://user-images.githubusercontent.com/45943148/101280173-e2eaf280-37c7-11eb-8377-3833d72e2725.png">

Is there a better way to load all the data at once so that no delays and loading times are to be expected throughout the experiment? Any hints would be appreciated. 

Have a great day, 
Christina

**Comment by jodeleeuw** [comment, marked as answer]
Hi @christinalubinus,

The 404 error in the network tab suggests that the files aren't in the place that the experiment is looking for them. I'd suggest double checking the paths and directories to make sure that the files are where you intend them to be.

**Reply by christinalubinus** [reply]
Hi @jodeleeuw,
Thanks for getting back to me! That is what I thought in the beginning as well. But when I reduce the amount of trials in the stimulus file, without changing any code or the location of the files, the experiment runs through smoothly. This is why I don't think it is the directories but rather an issue related to the amount of data to be loaded.

**Reply by jodeleeuw** [reply]
Can you confirm that a file that shows a 404 error when you have a large number doesn't show it when you have a small number? I wonder if some of the files are in the right place and others aren't. Sorry if this sounds like stuff you've already tried, but the 404 error is pretty specific to cases where the file can't be located so I'm trying to rule out the usual explanations.

**Reply by christinalubinus** [reply]
Wow, now I feel stupid. In one step of the processing pipeline of the stimuli, a "." was exchanged by a "_", which made audio filename and name on the stimulus list incompatible. And this just affected part of the stimuli. 
Thanks a lot @jodeleeuw!
